pyproject.toml

```diff
@@ -1,6 +1,98 @@
 [build-system]
-requires = ["setuptools>=51.0.0", "wheel"]
-build-backend = "setuptools.build_meta"
+requires = ["hatchling"]
+build-backend = "hatchling.build"
+
+[project]
+name = "odc-stac"
+description = "Tooling for converting STAC metadata to ODC data model"
+version = "0.4.0a1"
+authors = [
+    {name = "Open Data Cube"}
+]
+maintainers = [
+    {name = "Open Data Cube"}
+]
+readme = "README.rst"
+license = "Apache-2.0"
+requires-python = ">=3.8"
+classifiers = [
+    "License :: OSI Approved :: Apache Software License",
+    "Intended Audience :: Developers",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+    "Topic :: Scientific/Engineering :: GIS",
+    "Typing :: Typed"
+]
+dependencies = [
+    "affine",
+    "odc-geo>=0.4.7",
+    "odc-loader>=0.5.0",
+    "rasterio>=1.0.0,!=1.3.0,!=1.3.1",
+    "dask[array]",
+    "numpy>=1.20.0",
+    "pandas",
+    "pystac>=1.0.0,<2",
+    "toolz",
+    "xarray>=0.19"
+]
+
+[project.optional-dependencies]
```

Comment:
At some point, this section would be better represented as [dependency-groups](https://packaging.python.org/en/latest/specifications/dependency-groups/), which are for development dependencies, and not included in distribution packages.

If using `uv run/sync` the `dev` group is installed by default, but requires the `--group <name>` flag for others.

 It was only standardised in October 2024 so I'm not sure about support in `pip` or other tools.